WalletWasabi.Tests/UnitTests/Clients/SingleInstanceCheckerTests.cs

```diff
@@ -11,45 +14,138 @@ namespace WalletWasabi.Tests.UnitTests.Clients
+			await sicRegTest.EnsureSingleOrThrowAsync();
+			await using SingleInstanceChecker sicRegTest2 = new(regTestPort);
+			await Assert.ThrowsAsync<OperationCanceledException>(async () => await sicRegTest2.EnsureSingleOrThrowAsync());
+		}
+
+		[Fact]
+		public async Task OtherInstanceStartedTestsAsync()
+		{
+			int mainNetPort = GenerateRandomPort();
+
+			// Disposal test.
+			await using SingleInstanceChecker firstInstance = new(mainNetPort);
+			long eventCalled = 0;
+
+			firstInstance.OtherInstanceStarted += SetCalled;
+
+			try
 			{
-				await sic.CheckAsync();
-				await Assert.ThrowsAsync<InvalidOperationException>(async () => await sic.CheckAsync());
+				// I am the first instance this should be fine.
+				await firstInstance.EnsureSingleOrThrowAsync();
+
+				await using SingleInstanceChecker secondInstance = new(mainNetPort);
 
-				using SingleInstanceChecker sicMainNet2 = new SingleInstanceChecker(Network.Main, mainLockName);
-				await Assert.ThrowsAsync<InvalidOperationException>(async () => await sicMainNet2.CheckAsync());
+				for (int i = 0; i < 2; i++)
+				{
+					// I am the second one.
+					await Assert.ThrowsAsync<OperationCanceledException>(async () => await secondInstance.EnsureSingleOrThrowAsync());
+				}
 
-				string testnetLockName = GenerateLockName(Network.TestNet);
-				using SingleInstanceChecker sicTestNet = new SingleInstanceChecker(Network.TestNet, testnetLockName);
-				await sicTestNet.CheckAsync();
-				await Assert.ThrowsAsync<InvalidOperationException>(async () => await sicTestNet.CheckAsync());
+				// Overall Timeout.
+				using CancellationTokenSource cts = new(TimeSpan.FromSeconds(30));
 
-				string regtestLockName = GenerateLockName(Network.RegTest);
-				using SingleInstanceChecker sicRegTest = new SingleInstanceChecker(Network.RegTest, regtestLockName);
-				await sicRegTest.CheckAsync();
-				await Assert.ThrowsAsync<InvalidOperationException>(async () => await sicRegTest.CheckAsync());
+				// Simulate a port scan operation.
+				using (TcpClient client = new TcpClient()
+				{
+					NoDelay = true
+				})
+				{
+					// Make sure non cancel-able operations are aborted.
+					using var _ = cts.Token.Register(() => client.Dispose());
+					// This should not be counted.
+					await client.ConnectAsync(IPAddress.Loopback, mainNetPort, cts.Token);
+					await using NetworkStream networkStream = client.GetStream();
+					networkStream.WriteTimeout = (int)SingleInstanceChecker.ClientTimeOut.TotalMilliseconds;
+					await using var writer = new StreamWriter(networkStream, Encoding.UTF8);
+					await writer.WriteAsync(new StringBuilder("fake message"), cts.Token);
+				}
+
+				// Simulate a port scan operation.
+				try
+				{
+					using TcpClient client = new TcpClient()
+					{
+						NoDelay = true
+					};
+
+					// Make sure non cancel-able operations are aborted.
+					using var _ = cts.Token.Register(() => client.Dispose());
```

Comment:
Why to dispose `client` when there is `using` on line 104?

Reply:
I removed all of these patterns and used WithAwaitCancellation instead.